sql/mysqld.cc

```diff
@@ -6305,6 +6378,11 @@ void handle_connections_sockets()
       }
       continue;
     }
+    else if (retval == 0)
+    {
+      check_upgrade();
```

Comment:
I really don't like the idea of a constant nagging log entry. The server logs are meant to be minimal.

Reply:
I guess the alternative is to just print the message once when the server starts, that would reduce the amount of log messages but it will make the message easier to miss.

Reply:
So do you want me to remove the periodic check? I think it would be useful to keep printing the warning message as long as the upgrade tool haven't been executed, if something goes wrong during the upgrade process I think that would help to notice it as soon as possible. Note that once the `mysql_upgrade` tool is executed successfully and the upgrade file is properly set no warning or message will be printed.

Reply:
All other server warnings are pretty much designed to be issued once during the server uptime. (e.g. innodb stats missing, event hander table structure). So yes, remove the periodic check. With the in-server upgrade this eventually will become more meaningful as it will only occur if skipped.

Reply:
@grooverdan after some thoughts on this I still think that this warning should be periodic, note that the warning will appear always the first time the server is started with the new binaries after a major upgrade, and it should disappear once the `mysql_upgrade` has finished successfully and the `mysql_upgrade_file` contains the new version. So just printing the warning once when it starts doesn't makes sense unless the server is restarted after the upgrade, which AFAIK is not necessary.

Probably we could change that once we have the in-server upgrade. With the in-server upgrade, the server itself will be aware of the status of the upgrade and probably just printing either a success or an error message when the upgrade process is finished will be enough.

Reply:
Monty's objecting to the in-server upgrade - if you've got some great words can you add them to MDEV-30499 with a focus on why the current out of band is deficient and why multiple restarts is a bad idea.

Reply:
Sure, I can add a comment with my opinion in the JIRA.

Also, did you have time to think about my comments on the periodic change? do you think it makes sense to include it based on that?

Reply:
Hi @christian7007, whilst I understand your point of view, it goes against the philosophy behind the log. It will be very easy to miss other important entries if this constantly repeats. We would not be able to merge it with that.

Reply:
Hi @LinuxJedi, note that the goal of the periodic warning is indeed to make "enough noise" to make sure that, when needed, the `mariadb-upgrade` tool is executed ASAP and that the warning won't be printed again once the tool is executed successfully.

Reply:
Absolutely, I thoroughly understand the goal. The problem is that this is not the way to go about it. We cannot flood the error log with messages.

If noise is required, then having an external script probe for this state would likely be the way to do it.